extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php

```diff
@@ -70,6 +70,13 @@ public static function parserUpgradeSpecialFilePathURLs( string &$url ) {
 	 * @return boolean
 	 */
 	public static function onLinkerMakeExternalLink( string &$url, string &$text, bool &$link, array &$attribs ): bool {
+		global $wgFandomBaseDomain;
+
+		$host = parse_url( $url, PHP_URL_HOST );
+		$hostRegex = '/^([a-zA-Z0-9]+\.){0,1}'. $wgFandomBaseDomain .'/';
```

Comment:
I can see few issues here:
1. It does not handle wikia.org
2. I think we can safely assume all fandom.com and wikia.org links are https-compatible
3. Most likely it will not work for preview/verify/devboxes/sandboxes